src/components/LiquidityPoolInteraction.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from "react";
+import { Icon } from "@components/Icon";
+import { Button } from "@components/Button";
+import { Input } from "@components/Input";
+import { InvestmentDetails } from "@components/InvestmentDetails";
+import { Form } from "antd";
+import { FormProps } from "antd/lib/form";
+
+interface LiquidityPoolInteractionProps {
+	/**
+	 * The action that will be performed on the liquidity pool.
+	 * "invest" will transfer USDC from the user's wallet to the liquidity pool in exchange for LP tokens
+	 * "withdraw" will transfer USDC from the liquidity pool in exchange for LP tokens
+	 */
+	action: "invest" | "withdraw";
+	/**
+	 * Action to perform when form is submitted
+	 */
+	onSubmit: FormProps["onFinish"];
+	/**
+	 * Action to perform when form submission fails
+	 */
+	onSubmitFailed: FormProps["onFinishFailed"];
+}
+
+export const LiquidityPoolInteraction = ({
+	action,
+	onSubmit,
+	onSubmitFailed,
+}: LiquidityPoolInteractionProps) => {
+	// TODO: get these values from client
+	const balance = 65;
+	const investments = 256;
+	const investmentsReturn = 3.24;
+
+	const [form] = Form.useForm();
+	const [submitDisabled, setSubmitDisabled] = useState(null);
+
+	const onAddMax = () => {
+		form.setFieldsValue({ amount: action === "invest" ? balance : investments });
+		setSubmitDisabled(false);
+	};
+
+	// Set the disabled state on the client so there is no client <-> server mismatch
+	useEffect(() => setSubmitDisabled(true), []);
+
+	return (
+		<div className="p-6 md:p-12 bg-neutral-0 space-y-7">
+			<h2 className="space-x-5 flex items-center">
+				<Icon name="line-chart" className="w-7 h-7" />
+				<span className="uppercase font-bold text-2xl">{action}</span>
+			</h2>
+			<InvestmentDetails
+				balance={balance}
+				balanceCurrency="USDC"
+				investments={investments}
+				investmentsCurrency="USDC"
+				investmentsReturn={investmentsReturn}
+			/>
+			<Form
+				name="invest"
+				form={form}
+				onFinish={onSubmit}
+				onFinishFailed={onSubmitFailed}
+				layout="vertical"
+				onFieldsChange={(_changedFields, fields) =>
```

Comment:
create a onFieldsChange function

Reply:
it would be better if we only enable the submit button if the form validation succeeds instead of applying our own validation on top of the one we have for the form itself